Multi region

dtarb opened the pull request:
Multiregion capability to merge

dtarb committed:
Add ability to process multiple regions and integrate

dtarb committed:
Add pycharm .idea folder to .gitignore

dtarb committed:
Change to delineate multiple NHDPlus regions

dtarb committed:
Fixed gage watershed grid type (32 bit unsigned), preprocessing for internally draining regions

dtarb committed:
Add use of modified flow direction input file fdrmod.tif where it is present in the region directory.

pkdash commented on DoMergeArcPro.py:
```diff
@@ -0,0 +1,130 @@
+    downcatch = 2
+    with open(masterIDfile) as f:
+        lines = f.read().splitlines()
+    for x in lines:
```
Seems this ```for``` loop should be inside ```with```

dtarb replied:
Sure - will do.

pkdash commented on DoMergeArcPro.py:
```diff
@@ -0,0 +1,130 @@
+    arcpy.management.Dissolve(tempoutput,output)
+
+def read_regionlist(input_dir_name, regionlistfile):
+    with open(os.path.join(input_dir_name, regionlistfile), 'r') as f:
```
Would it be better to document the format of the data in the ```regionlistfile``` here?

dtarb replied:
Sure.

pkdash commented:
@dtarb I have now looked at this PR. Mostly there are python styling issues. If you want I can clean that up.

dtarb committed:
Fix indefinite recursion when flow directions are inconsistent where regions overlap
Add 1 m buffer in merging of watersheds to eliminate slivers
Add zip to ship function to zip distributable files

dtarb commented:
We have to be careful merging this.  The code on the multiregion branch is for multiple regions from NHDPlus.  We need a separate place/folder for the preprocessing when it is not NHDPlus and there are not multiple regions involved.  I am not sure if the master branch holds this, but if it does we need to not overwrite this.

dtarb committed:
Document regionlist file format and improve style.

dtarb commented:
@pkdash I checked the master branch and it does not hold any non-NHDPlus preprocessing.  So this is ready to merge.  Please either give this a +1 or merge.

pkdash commented:
+1

dtarb committed:
Correct preprocess2.bat to use total upstream length (tlen) from gridnet on stream raster mask (src) to get Drainage Density right.
